ui/src/shared/components/FillQuery.tsx

```diff
@@ -37,6 +38,28 @@ class FillQuery extends PureComponent<Props, State> {
     value: NULL_STRING,
   }
 
+  public static getDerivedStateFromProps(props: Props, state: State) {
+    if (state.value === props.value) {
+      return false
+    }
+
+    const isNumberValue: boolean = !isNaN(Number(props.value))
+
+    if (isNumberValue) {
+      return {
+        selected: queryFills.find(fill => fill.type === NUMBER),
+        currentNumberValue: props.value,
+        resetNumberValue: props.value,
+        value: props.value,
+      }
+    }
+
+    return {
+      selected: queryFills.find(fill => fill.type === props.value),
```

Comment:
could the function used to find selected ever not find something?

Reply:
yes but a null value should just mean that the first item in the list is selected